spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/aws/KinesisService.java

```diff
@@ -30,7 +30,13 @@ public class KinesisService {
 
     public KinesisService() {
         // AWS web identity is set for EKS clusters, if these are not set then use default credentials
-        if (System.getenv("AWS_WEB_IDENTITY_TOKEN_FILE") == null && System.getProperty("aws.webIdentityTokenFile") == null) {
+        if (System.getenv("AWS_DEFAULT_REGION") != null) {
```

Comment:
Is `AWS_DEFAULT_REGION` guaranteed to be set? In ECS only?

Reply:
In ECS task container for customers service, `AWS_DEFAULT_REGION` is specified ([Ref](https://github.com/blairhyy-amazon/application-signals-demo/blob/0213b51dd4d42e765a084a12827f5631243ad2a0/cdk/ecs/lib/stacks/ecsStack.ts#L521)) so that `KinesisService` and `SqsService` can be deployed in the correct region. Same for `DdbService` in visits service.